### PR TITLE
Changed Analytics-Worker kind to StatefulSet

### PR DESCRIPTION
**Pull request by BLasan**
**Changes**

- Deployed Analytics-Worker as a StatefulSet
- Added updateApiMangerStatusForStatefulSet(param1, param2) for updating status

### CODE REVIEW

**Review by pubudu538 — APPROVED** [review]

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -529,12 +523,15 @@ func (c *Controller) syncHandler(key string) error {
 		}
 
 		// Get analytics worker deployment name using hardcoded value
-		workerdeployment, err := c.deploymentsLister.Deployments(apimanager.Namespace).Get(workerDeploymentName)
+
+		// workerdeployment, err := c.deploymentsLister.Deployments(apimanager.Namespace).Get(workerDeploymentName)
```
Comment (SanojPunchihewa) [inline pkg/controller/controller.go:527]: Shall we remove these commented lines? Check line 533 as well

Comment (BLasan) [reply pkg/controller/controller.go:527]: Commented lines do not exist in the master branch

Comment (SanojPunchihewa) [reply pkg/controller/controller.go:527]: Seems like the line numbers are different in master https://github.com/wso2/k8s-wso2am-operator/blob/master/pkg/controller/controller.go#L517

Comment (BLasan) [reply pkg/controller/controller.go:527]: fixed via https://github.com/wso2/k8s-wso2am-operator/pull/90/commits/a576020688d42ebe8201635fa8d9689cffe0437e

**Pull request merged**